config/config.go

```diff
@@ -32,3 +42,39 @@ func AddFlags(cmd *cobra.Command) {
 		panic(err)
 	}
 }
+
+func PersistentStartupProcessFlags() error {
+
+	if err := convertResolveHostsToMap(); err != nil {
+		return err
+	}
+	return nil
+
+}
+
+func convertResolveHostsToMap() error {
+	for _, resolveHost := range viper.GetStringSlice(optname.Resolve) {
+		split := strings.SplitN(resolveHost, ":", 3)
+		if len(split) != 3 {
+			return fmt.Errorf("invalid resolve host format, expected <hostname>:port:<ip>, got: %s", resolveHost)
+		}
+		host, port, addr := strings.ToLower(split[0]), split[1], split[2]
+		if _, ok := HostToIPResolutionMap[host]; ok {
```

Comment:
we don't store `host` in the map, so `ok` won't ever be true here

Reply:
Hmm. It looks like this code shifted between two branches. Let me re-look at it. This is a mis-mash of two distinct iterations. Thats why this all doesn't line up.